app/src/main/java/com/trainstation/frettirtest2/MyViewPagerFragment.java

```diff
@@ -50,8 +42,10 @@ public void onCompleted() {
 
             @Override
             public void onError(Throwable e) {
-                // TODO: show toast
-                Log.e("MyViewPagerFragment", "Couldn't get response", e);
+                Context context = getContext();
+                Toast.makeText(context, context.getResources().getString(R.string.error_getting_news), Toast.LENGTH_LONG)
+                        .show();
```

Comment:
there was a TODO in the code to show a toast, so I added that,

`makeText()` has three parameters
- **Context context** I got that here https://github.com/helgio/frettaApp/pull/2/files#r51474942
- **CharSequence text** note that you can use String when CharSequence is needed. so instead of hardcoding some string I used `context.getResources().getString(R.string.error_getting_news)` so what is happening here? --> when adding string to a User Interface (UI), something that the user will see, then Android recommend to declare the hardcoded string in R.string file, see: https://github.com/helgio/frettaApp/pull/2/files#diff-01dafb3fd0217441330103cfc8300094R3, but note that `R.string.error_getting_news` is just an int, it is a int that reference some number that reference the string or something like that, so we need to add `getString()` around that and then to get the string we need to "load" all the resources (by that I mean everything under the res/R folder) by adding `context.getResources()` in front. I don't understand completely what that is needed, but when you need to access in code (not xml/layout file) you have to do this, something that we should just remember :)
- **int duration**, you can see that I use `Toast.LENGTH_LONG` (which is actually a int) which is declared in the `Toast ` class, and it is declared there as a option for the duration, and one of the reason for that is to have some name for that, e.g, the word `LENGTH_LONG` is much self-explanated compared to just the number `1`